example/src/main/java/com/basistheory/android/example/MainActivity.kt

```diff
@@ -13,34 +16,67 @@ import org.threeten.bp.Instant
 import org.threeten.bp.temporal.ChronoUnit
 
 class MainActivity : AppCompatActivity() {
+    private lateinit var cardNumberElement: CardNumberElement
     private lateinit var nameElement: TextElement
     private lateinit var phoneNumberElement: TextElement
     private lateinit var socialSecurityNumberElement: TextElement
     private lateinit var orderNumberElement: TextElement
     private lateinit var tokenizeResult: TextView
+    private lateinit var tokenizeButton: Button
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
 
+        cardNumberElement = findViewById(R.id.cardNumber)
         nameElement = findViewById(R.id.name)
         phoneNumberElement = findViewById(R.id.phoneNumber)
         socialSecurityNumberElement = findViewById(R.id.socialSecurityNumber)
         orderNumberElement = findViewById(R.id.orderNumber)
         tokenizeResult = findViewById(R.id.tokenizeResult)
+        tokenizeButton = findViewById(R.id.tokenizeButton)
 
         val digitRegex = Regex("""\d""")
         val charRegex = Regex("""[A-Za-z]""")
 
-        phoneNumberElement.keyboardType = KeyboardType.NUMBER // illustrates that it can be set programmatically
-        phoneNumberElement.mask = listOf("+", "1", "(", digitRegex,digitRegex,digitRegex, ")", " ", digitRegex, digitRegex, digitRegex, "-", digitRegex, digitRegex , digitRegex, digitRegex )
+        phoneNumberElement.keyboardType = KeyboardType.NUMBER
+        phoneNumberElement.mask = listOf(
+            "+",
+            "1",
+            "(",
+            digitRegex,
+            digitRegex,
+            digitRegex,
+            ")",
+            " ",
+            digitRegex,
+            digitRegex,
+            digitRegex,
+            "-",
+            digitRegex,
+            digitRegex,
+            digitRegex,
+            digitRegex
+        )
 
-        orderNumberElement.mask = listOf(charRegex, charRegex, charRegex, "-", digitRegex, digitRegex, digitRegex)
+        orderNumberElement.mask =
+            listOf(charRegex, charRegex, charRegex, "-", digitRegex, digitRegex, digitRegex)
+
+        cardNumberElement.addChangeEventListener {
+            if (!it.isValid && it.isComplete) {
+                cardNumberElement.textColor = Color.RED
+                tokenizeButton.isEnabled = false
+            } else {
+                cardNumberElement.textColor = Color.BLACK
+                tokenizeButton.isEnabled = true
+            }
+        }
```

Comment:
I almost deleted this, but I left it for now thinking we can move it into another activity later to demo validation